doc/source/gapfill.md

```diff
@@ -0,0 +1,272 @@
+# Gap-filling
+
+xDEM contains routines to gap-fill elevation data or elevation differences depending on the type of terrain.
+
+```{important}
+Some of the approaches below are application-specific (e.g., glaciers) and might be moved to a separate package
+in future releases.
+```
+
+So far, xDEM has three types of gap-filling methods:
+
+- Linear spatial interpolation,
+- Local hypsometric interpolation (only relevant for elevation differences and glacier applications),
+- Regional hypsometric interpolation (also for glaciers).
+
+The last two methods are described in [McNabb et al. (2019)](https://doi.org/10.5194/tc-13-895-2019).
+
+```{code-cell} ipython3
+:tags: [remove-cell]
+
+# To get a good resolution for displayed figures
+from matplotlib import pyplot
+pyplot.rcParams['figure.dpi'] = 600
+pyplot.rcParams['savefig.dpi'] = 600
+pyplot.rcParams['font.size'] = 9  # Default 10 is a bit too big for coregistration plots
+```
+
+```{code-cell} ipython3
+:tags: [hide-cell]
+:mystnb:
+:  code_prompt_show: "Show the code for opening example data"
+:  code_prompt_hide: "Hide the code for opening example data"
+
+from datetime import datetime
+
+import geoutils as gu
+import numpy as np
+import matplotlib.pyplot as plt
+
+import xdem
+
+# Load a reference DEM from 2009
+dem_2009 = xdem.DEM(xdem.examples.get_path("longyearbyen_ref_dem"), datetime=datetime(2009, 8, 1))
+# Load a DEM from 1990
+dem_1990 = xdem.DEM(xdem.examples.get_path("longyearbyen_tba_dem"), datetime=datetime(1990, 8, 1))
+# Load glacier outlines from 1990.
+glaciers_1990 = gu.Vector(xdem.examples.get_path("longyearbyen_glacier_outlines"))
+glaciers_2010 = gu.Vector(xdem.examples.get_path("longyearbyen_glacier_outlines_2010"))
+
+# Make a dictionary of glacier outlines where the key represents the associated date.
+outlines = {
+    datetime(1990, 8, 1): glaciers_1990,
+    datetime(2009, 8, 1): glaciers_2010,
+}
+
+# Cropping DEMs to a smaller extent to visualize the gap-filling better
+bounds = (dem_2009.bounds.left, dem_2009.bounds.bottom,
+          dem_2009.bounds.left + 200 * dem_2009.res[0], dem_2009.bounds.bottom + 150 * dem_2009.res[1])
+dem_2009 = dem_2009.crop(bounds)
+dem_1990 = dem_1990.crop(bounds)
+```
+
+We create a difference of DEMs object {class}`xdem.ddem.dDEM` to experiment on:
+
+```{code-cell} ipython3
+ddem = xdem.dDEM(raster=dem_2009 - dem_1990, start_time=dem_1990.datetime, end_time=dem_2009.datetime)
+
+# The example DEMs are void-free, so let's make some random voids.
+# Introduce 50000 nans randomly throughout the dDEM.
+mask = np.zeros_like(ddem.data, dtype=bool)
+mask.ravel()[(np.random.choice(ddem.data.size, int(ddem.data.size/5), replace=False))] = True
+ddem.set_mask(mask)
+```
+
+## Linear spatial interpolation
+
+Linear spatial interpolation (also often called bilinear interpolation) of dDEMs is arguably the simplest approach: voids are filled by an average of the surrounding pixels values, weighted by their distance to the void pixel.
```

Comment:
Actually it is not a bilinear interpolation, but an Inverse Distance Weighting interpolation. Bilinear would be to interpolate between 4 pixels. At least according to wikipedia: "Bilinear interpolation considers the closest 2 × 2 neighborhood of known pixel values surrounding the unknown pixel's computed location."

Reply:
That's an important detail! Changing everywhere.